src/ui/public/directives/input_number.js

```diff
@@ -0,0 +1,18 @@
+import uiModules from 'ui/modules';
+let module = uiModules.get('kibana');
+
+module.directive('inputNumber', function () {
+  return {
+    restrict: 'A',
+    require: 'ngModel',
+    link: function ($scope, $elem, attrs, ngModel) {
+      ngModel.$parsers.push(checkNumber);
```

Comment:
Not sure regarding official style either, I was just emulating how it was done in the input_whole_number.js